ssl_test.go

```
@@ -98,83 +98,56 @@ JBfFkFXDvcbaYmpcOVCS1susPrPr8rgIm+vK6X+UoWE1RcCMGQMKObQIDGpE4IWe
-FhlGM1wzvusyGrm26Vrbqm4wDwYDVR0TAQH/BAUwAwEB/zAKBggqhkjOPQQDAgNJ
-ADBGAiEA6PWNjm4B6zs3Wcha9qyDdfo1ILhHfk9rZEAGrnfyc2UCIQD1IDVJUkI4
-J/QVoOtP5DOdRPs/3XFy0Bk0qH+Uj5D7LQ==
------END CERTIFICATE-----
-`)
-	ed25519CertBytes = []byte(`-----BEGIN CERTIFICATE-----
-MIIBIzCB1gIUd0UUPX+qHrSKSVN9V/A3F1Eeti4wBQYDK2VwMDYxCzAJBgNVBAYT
-AnVzMQ0wCwYDVQQKDARDU0NPMRgwFgYDVQQDDA9lZDI1NTE5X3Jvb3RfY2EwHhcN
-MTgwODE3MDMzNzQ4WhcNMjgwODE0MDMzNzQ4WjAzMQswCQYDVQQGEwJ1czENMAsG
-A1UECgwEQ1NDTzEVMBMGA1UEAwwMZWQyNTUxOV9sZWFmMCowBQYDK2VwAyEAKZZJ
-zzlBcpjdbvzV0BRoaSiJKxbU6GnFeAELA0cHWR0wBQYDK2VwA0EAbfUJ7L7v3GDq
-Gv7R90wQ/OKAc+o0q9eOrD6KRYDBhvlnMKqTMRVucnHXfrd5Rhmf4yHTvFTOhwmO
-t/hpmISAAA==
------END CERTIFICATE-----
-`)
-	ed25519KeyBytes = []byte(`-----BEGIN PRIVATE KEY-----
-MC4CAQAwBQYDK2VwBCIEIL3QVwyuusKuLgZwZn356UHk9u1REGHbNTLtFMPKNQSb
------END PRIVATE KEY-----
-`)
+`
 )
 
 // NetPipe creates a TCP connection pipe and returns two connections.
-func NetPipe(t testing.TB) (net.Conn, net.Conn) {
-	l, err := net.Listen("tcp", "localhost:0")
+func NetPipe(tb testing.TB) (net.Conn, net.Conn) {
+	tb.Helper()
+
+	lis, err := net.Listen("tcp", "localhost:0")
 	if err != nil {
-		t.Fatal(err)
+		tb.Fatal(err)
 	}
-	defer l.Close()
+	defer lis.Close()
 
 	// Use Future pattern to create client connection asynchronously
 	clientFuture := utils.NewFuture()
 	go func() {
-		clientFuture.Set(net.Dial(l.Addr().Network(), l.Addr().String()))
+		clientFuture.Set(net.Dial(lis.Addr().Network(), lis.Addr().String()))
 	}()
 
-	var errs utils.ErrorGroup
-	serverConn, err := l.Accept()
+	var (
+		errs utils.ErrorGroup
+		conn net.Conn
+		ok   bool
+	)
+
+	serverConn, err := lis.Accept()
 	errs.Add(err)
 	clientConn, err := clientFuture.Get()
 	errs.Add(err)
 
+	if clientConn != nil {
+		conn, ok = clientConn.(net.Conn)
+	}
+
 	err = errs.Finalize()
 	if err != nil {
 		if serverConn != nil {
-			err := serverConn.Close()
-			if err != nil {
-				t.Fatal(err)
-			}
+			_ = serverConn.Close()
```

Why ignore the error return by Close() ?

---

Fixed